api-put-object.go

```diff
@@ -205,10 +205,13 @@ func (c Client) putObjectSingle(bucketName, objectName string, reader io.Reader,
 		hashAlgos["sha256"] = sha256.New()
 	}
 
-	if size <= minPartSize {
+	if size <= minPutBufferSize {
 		// Initialize a new temporary buffer.
 		tmpBuffer := new(bytes.Buffer)
 		size, err = hashCopyN(hashAlgos, hashSums, tmpBuffer, reader, size)
+		if err != nil {
```

Comment:
Actually, hashCopyN can return io.EOF in normal conditions (not sure why), besides, there is a check of err later.